backend/src/service/oia/oianalytics-registration.service.ts

```diff
@@ -138,6 +138,35 @@ export default class OIAnalyticsRegistrationService {
     this.oIAnalyticsRegistrationRepository.updateKeys(await encryptionService.encryptText(privateKey), publicKey);
   }
 
+  async testConnection(command: RegistrationSettingsCommandDTO): Promise<void> {
+    await this.validator.validate(registrationSchema, command);
+
+    const currentRegistration = this.oIAnalyticsRegistrationRepository.get()!;
+
+    // Prepare proxy password - use existing if not provided
+    let proxyPassword = command.proxyPassword;
+    if (!proxyPassword) {
+      proxyPassword = currentRegistration.proxyPassword;
+    } else {
+      proxyPassword = await encryptionService.encryptText(command.proxyPassword);
+    }
+
+    // Import the test connection utility
+    const { testOIAnalyticsConnection } = await import('./oianalytics-test-connection.utils');
```

Comment:
Please stick to our code standards: no dynamic import. Also adapt the test code to match the standards